r/src/type_infer.cpp

```diff
@@ -72,7 +72,8 @@ std::shared_ptr<arrow::DataType> InferArrowTypeFromVector<INTSXP>(SEXP x) {
   } else if (Rf_inherits(x, "POSIXct")) {
     auto tzone_sexp = Rf_getAttrib(x, symbols::tzone);
     if (Rf_isNull(tzone_sexp)) {
```

Comment:
```suggestion
    if (Rf_isNull(tzone_sexp) || tzone_sexp == "") {
```

Reply:
A naive comparison with an empty string does not work. I think I might need to read up on that. I think I need to somehow declare + initialise the empty string.